src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -341,6 +341,72 @@ public function symlink($originDir, $targetDir, $copyOnWindows = false)
     }
 
     /**
+     * Return the next direct target of a link.
+     *
+     * To recursively follow links until a final target
+     * is reached, use Filesystem::realpath($path).
+     *
+     * @param string $path A link path.
+     *
+     * @return string Return the resolved link.
+     *
+     * @throws IOException When the link does not exist or is not readable.
+     */
+    public function readlink($path)
+    {
+        if (!$this->exists($path)) {
+            throw new IOException(sprintf('The link %s does not exist and cannot be read.', $path));
+        }
+
+        if (!is_link($path)) {
+            throw new IOException(sprintf('The path %s is not a link.', $path));
+        }
+
+        // /link1 -> /link2 -> /file
+
+        // Windows: readlink(/link1) => /file
+        //          realpath(/link1) => /link2
+
+        // Unix:    readlink(/link1) => /link2
+        //          realpath(/link1) => /file
+
+        if ('\\' === DIRECTORY_SEPARATOR) {
+            return realpath($path);
+        }
+
+        return readlink($path);
+    }
+
+    /**
+     * Return the final target of a link by recursively following links.
+     *
+     * To find only the direct next target of a link,
+     * use Filesystem::readlink($path).
+     *
+     * @param string $path A link path.
+     *
+     * @return string Return the final target of the link.
+     *
+     * @throws IOException When the link does not exist or is not readable.
+     */
+    public function realpath($path)
+    {
+        if (!$this->exists($path)) {
+            throw new IOException(sprintf('The link %s does not exist and cannot be read.', $path));
+        }
+
+        if (!is_link($path)) {
```

Comment:
I'm not sure what behavior we should have. The native behavior is to return false, but that seems inapropriate to me as we should return a string. What would you expect?